File: 2_ReactWeb/src/entities/llm-provider/model/discoveredModel.ts
export type DiscoveredModelCapability =
  | "reasoning"
  | "vision"
  | "websearch"
  | "embedding"
  | "rerank"
  | "function_calling";

export interface DiscoveredModelEntry {
  model_id: string;
  display_name: string;
  provider_id: string;
  family_group: string;
  capability_tags: DiscoveredModelCapability[];
}

export interface DiscoveredModelListResponse {
  count: number;
  items: DiscoveredModelEntry[];
}

export interface ProviderCloudModelCacheResponse extends DiscoveredModelListResponse {
  api_base_url: string;
  discovered_at: string | null;
  has_cache: boolean;
  protocol_family: string;
  provider_id: string;
}

File: 1_PythonServer/app/domain/llm/discovered_model.py
# 发现的模型领域模型
# 上游供应商模型发现接口返回的模型摘要信息

from dataclasses import dataclass


@dataclass(frozen=True, slots=True)
class DiscoveredModel:
    """发现的模型：ID、显示名称、所属供应商、族群分类和功能标签"""

    model_id: str
    display_name: str
    provider_id: str
    family_group: str = ""
    capability_tags: tuple[str, ...] = ()
    raw_payload: dict[str, object] | None = None
